### DISCUSSION TITLE
Get ecal status in python

### DISCUSSION THREAD

**Discussion by ENAC-Robotique** [eclipse-ecal/ecal#1087] [Q&A]
Hello, I'm using the ecal python bridge and I wonder how can I get the ecal status?

I am using ecal with network enabled and I have to set a multicast route for ecal to work, and since my program starts before the route is set (its a linux service), it can't connect at first. 
I want to test the status and quit the program is it's not Ok.

Ecal prints `CUDPSender::Send failed with: 'Network is unreachable'` in the terminal but I don't know what function to call to get this status ?

I suspect that `eCAL::Time::GetStatus` is the function I want, but there seems to be no python binding for this function.

NB: `ecal.core.core.ok()` returns 1 so it doesn't help for this case.

**Comment by KerstinKeller** [comment]
Maybe @rex-schilasky can answer this better.
In general, with `ecal.core.core.ok()` you can check if the initialization was ok. But this checks only internal initialization, and I do not think that this checks if routes are set correctly (after all, at least in local mode, this is not a requirement). So I don't think that there is a way to check with eCAL if your system is configured correctly.
It could be interesting, to add checks like this.

**Comment by rex-schilasky** [comment]
Yes, unfortunately `ecal.core.core.ok()` will not reflect the status of the eCAL network layer connection. I can imaging that you could use another system function to check if the network routes are set or not. Maybe you can utilize an python `os.system` call and parse its output ?

**Comment by ENAC-Robotique** [comment]
Since ecal is printing some errors I though it would be more reliable than reading system config, that could be erroneous.
I will try doing a test in C++  to test if the GetStatus function can help with it.

**Comment by ENAC-Robotique** [comment]
I can confirm eCAL::Time::GetStatus is not want I want :smile:  (I printed the error code ans status message)
```
CUDPReceiverAsio: Unable to join multicast group: No such device
CUDPSender::Send failed with: 'Network is unreachable'
CUDPReceiverAsio: Unable to join multicast group: No such device
0   everything is fine.
0   everything is fine.
CUDPSender::Send failed with: 'Network is unreachable'
CUDPSender::Send failed with: 'Network is unreachable'
0   everything is fine.
0   everything is fine.
```

**Comment by ENAC-Robotique** [comment]
I tracked down the message to the AddMultiCastGroup method, and I see that the return value is not used. Maybe it could be a good idea to use this return value?
https://github.com/eclipse-ecal/ecal/blob/534bc59ea19ce7a4e43c21e257420b14b362d160/ecal/core/src/io/udp_receiver_asio.cpp#L100-L103

**Reply by rex-schilasky** [reply]
We will check this. Thank you for your effort.
